Support referring interface via PCI-address

cathay4t opened the pull request:
Introducing `identifier: pci-address` and `pci-address` property to
`interface` section for matching interface by PCI-address. For example:

```yml
---
interfaces:
  - name: wan0
    type: ethernet
    state: up
    identifier: pci-address
    pci-address: 0000:07:00.0
```

You can refer profile name in bond/bridge/vrf port or
VLAN/VxLAN/MacSec/MacVlan/MacVtap base-iface.
For example:

```yml
---
interfaces:
  - name: port1
    type: ethernet
    state: up
    identifier: pci-address
    pci-address: 0000:07:00.0
  - name: port1.100
    type: vlan
    vlan:
      base-iface: port1
      id: 100
  - name: bond0
    type: bond
    link-aggregation:
      mode: balance-rr
      port:
      - port1
```

Integration test cases included. Since this require real hardware to
test, they are marked as skip and will only been invoked with
system environment variable `TEST_REAL_NIC` defined. For example:

```bash
    env TEST_REAL_NIC=enp7s0 pytest
```

Resolves: https://issues.redhat.com/browse/RHEL-88993

bengal commented on rust/src/lib/ifaces/inter_ifaces.rs:
```diff
@@ -461,6 +461,73 @@ impl Interfaces {
+    // If any desired interface has `identifier: mac-address`:
+    //  * Resolve interface.name to MAC address match interface name.
+    //  * Store interface.name to interface.profile_name.
+    fn resolve_pci_identifider(
```
s/identifider/identifier/

cathay4t replied:
Fixed. Thanks.

cathay4t committed:
Support referring interface via PCI-address

Introducing `identifier: pci-address` and `pci-address` property to
`interface` section for matching interface by PCI-address. For example:

```yml
---
interfaces:
  - name: wan0
    type: ethernet
    state: up
    identifier: pci-address
    pci-address: 0000:07:00.0
```

You can refer profile name in bond/bridge/vrf port or
VLAN/VxLAN/MacSec/MacVlan/MacVtap base-iface.
For example:

```yml
---
interfaces:
  - name: port1
    type: ethernet
    state: up
    identifier: pci-address
    pci-address: 0000:07:00.0
  - name: port1.100
    type: vlan
    vlan:
      base-iface: port1
      id: 100
  - name: bond0
    type: bond
    link-aggregation:
      mode: balance-rr
      port:
      - port1
```

Integration test cases included. Since this require real hardware to
test, they are marked as skip and will only been invoked with
system environment variable `TEST_REAL_NIC` defined. For example:

```bash
    env TEST_REAL_NIC=enp7s0 pytest
```

Resolves: https://issues.redhat.com/browse/RHEL-88993

Signed-off-by: Gris Ge <fge@redhat.com>
